lib/ssl/src/ssl.erl

```diff
@@ -1910,6 +1923,29 @@ check_cert_key(UserOpts, CertKeys, LogLevel) ->
                 end,
     CertKeys2.
 
+check_key_algo_version_dep(Versions, eddsa) ->
+    assert_version_dep(key, Versions, ['tlsv1.3']);
+check_key_algo_version_dep(Versions, rsa_pss_pss) ->
+    assert_version_dep(key, Versions, ['tlsv1.3', 'tlsv1.2']);
+check_key_algo_version_dep(Versions, dsa) ->
+    assert_version_dep(key, Versions, ['tlsv1.2', 'tlsv1.1', 'tlsv1']);
+check_key_algo_version_dep(_,_) ->
+    true.
+
+check_key_legacy_version_dep(Versions, Key, rsa) ->
+    check_key_legacy_version_dep(Versions, Key);
+check_key_legacy_version_dep(_,_,_) ->
+    true.
+
+check_key_legacy_version_dep(Versions, Key) ->
+    EncryptFun = maps:get(encrypt_fun, Key, undefined),
+    case EncryptFun of
+        undefined ->
+            assert_version_dep(key, Versions, ['tlsv1.3', 'tlsv1.2']);
```

Comment:
Isn't undefined allowed in tlsv1.3 and tlsv1.2?

Reply:
If it is not defined we assert that the version needing it are not present!